python/ray/tune/ray_trial_executor.py

```diff
@@ -257,8 +262,16 @@ def _train(self, trial):
             return
 
         assert trial.status == Trial.RUNNING, trial.status
+        buffer_time_s = max(
+            TUNE_RESULT_BUFFER_MIN_TIME_S,
+            min(TUNE_RESULT_BUFFER_MAX_TIME_S,
+                len(self._running) // 10))
         with self._change_working_directory(trial):
-            remote = trial.runner.train.remote()
+            if TUNE_RESULT_BUFFER_LENGTH > 1:
+                remote = trial.runner.train_buffered.remote(
+                    buffer_time_s, TUNE_RESULT_BUFFER_LENGTH)
```

Comment:
I think we need the min between trial checkpoint frequency and buffer length right?

Also, could we add a test for that?

Reply:
Oh, good point, thanks for catching that. I added a unit test for this. (We can also include it in a release test)